src/main/java/com/mcmoddev/lib/capability/ICapabilitiesContainer.java

```diff
@@ -0,0 +1,36 @@
+package com.mcmoddev.lib.capability;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import mcp.MethodsReturnNonnullByDefault;
+
+/**
+ * A container to hold capabilities from multiple sources.
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
```

Comment:
Since these annotations seem to be the norm here, consider adding them to package-info.java?

Reply:
yes... problem is that in that package there are other classes not done by me... so... that will be @dshadowwolf 's call...

Reply:
Go for it - at this point we're trying to get things settled and I'd actually like to see those on every package. It'll make the code clearer and less cluttered.